resolver/resolver.cc

```diff
@@ -1731,6 +1731,12 @@ class ResolveTypeMembersAndFieldsWalk {
         } else if (cast->cast != core::Names::let()) {
             if (auto e = ctx.beginError(cast->loc, core::errors::Resolver::ConstantAssertType)) {
                 e.setHeader("Use `{}` to specify the type of constants", "T.let");
+                if (cast->cast == core::Names::cast()) {
+                    e.addErrorNote("If you really want to use `T.cast`, "
+                                   "assign to an intermediate variable first and then "
+                                   "assign that variable to `{}`",
+                                   asgn.lhs.toString(ctx));
+                }
```

Comment:
This seems reasonable to me, but what do you think about adding an autocorrect as well?

Reply:
I'm happy to add one.

Do we have a policy on adding autocorrects for things?  My assumption was that we only have autocorrects for things that are "obviously" the right thing to do, but adding an autocorrect here is more like "probably" the right thing to do.